### PR TITLE
Added basic support for pornhubpremium.com when using --cookies.

### PR DESCRIPTION
**Pull request by twaddington**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Ref #18797

Update the pornhub extractor to support pornhubpremium URLs. Throw an extractor error if --cookies is not provided.

### CODE REVIEW

File: youtube_dl/extractor/pornhub.py [hunk 1]
```diff
@@ -165,6 +165,14 @@ def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
         host = mobj.group('host') or 'pornhub.com'
         video_id = mobj.group('id')
+        cookie_file = self._downloader.params.get('cookiefile')
+
+        if host == 'pornhubpremium.com':
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Should also match `pornhubpremium.net`. Or extractor should not match `pornhubpremium.net`.

File: youtube_dl/extractor/pornhub.py [hunk 2]
```diff
@@ -165,6 +165,14 @@ def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
         host = mobj.group('host') or 'pornhub.com'
         video_id = mobj.group('id')
+        cookie_file = self._downloader.params.get('cookiefile')
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: Move under `if` where it's used.

File: youtube_dl/extractor/pornhub.py [hunk 3]
```diff
@@ -165,6 +165,14 @@ def _real_extract(self, url):
...
+
+        if host == 'pornhubpremium.com':
+            if not cookie_file:
+                raise ExtractorError(
+                    'PornHub Premium requires authentication.'
+                    ' You may want to use --cookies.',
```
Comment (dstftw) [inline youtube_dl/extractor/pornhub.py]: You must provide account credentials/cookies for testing.

Comment (twaddington) [reply youtube_dl/extractor/pornhub.py]: Sure thing. I emailed you a link to a gist.

File: youtube_dl/extractor/pornhub.py (hunk 1 above)
Comment (twaddington) [reply youtube_dl/extractor/pornhub.py]: Updated.

Where is `pornhub.net` used in the wild? For me it just redirects to `.com`.

File: youtube_dl/extractor/pornhub.py (hunk 2 above)
Comment (twaddington) [reply youtube_dl/extractor/pornhub.py]: Fixed.

File: youtube_dl/extractor/pornhub.py (hunk 1 above)
Comment (dstftw) [reply youtube_dl/extractor/pornhub.py]: Not quite sure. Currently it redirects to `pornhub.com`. Possibly this was not the case the past.

**Comment by twaddington** [comment]
@dstftw thanks for the quick review and merge!

**Comment by jordiaragonzaragoza** [comment]
@twaddington It does not work for me. I have opened an issue [https://github.com/ytdl-org/youtube-dl/issues/24404](url)

**Comment by gpy-dev** [comment]
Hello. I need your help! #24479

**Comment by gpy-dev** [comment]
> Hello. I need your help! #24479

Solved this problem.
https://github.com/ytdl-org/youtube-dl/issues/24479#issuecomment-604798156

**Comment by eromoe** [comment]
What is wrong ?
```
Traceback (most recent call last):
  File "d:\anaconda3\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "d:\anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "D:\Anaconda3\Scripts\youtube-dl.exe\__main__.py", line 9, in <module>
  File "d:\anaconda3\lib\site-packages\youtube_dl\__init__.py", line 474, in main
    _real_main(argv)
  File "d:\anaconda3\lib\site-packages\youtube_dl\__init__.py", line 441, in _real_main
    with YoutubeDL(ydl_opts) as ydl:
  File "d:\anaconda3\lib\site-packages\youtube_dl\YoutubeDL.py", line 419, in __init__
    self._setup_opener()
  File "d:\anaconda3\lib\site-packages\youtube_dl\YoutubeDL.py", line 2329, in _setup_opener
    self.cookiejar.load(ignore_discard=True, ignore_expires=True)
  File "d:\anaconda3\lib\site-packages\youtube_dl\utils.py", line 2762, in load
    self._really_load(cf, filename, ignore_discard, ignore_expires)
  File "d:\anaconda3\lib\http\cookiejar.py", line 2073, in _really_load
    (filename, line))
http.cookiejar.LoadError: invalid Netscape format cookies file 'C:\\Users\\Kasim\\youtube_dl_cookies\\pornhub.txt': 'www.pornhubpremium.com  FALSE
   /   FALSE   0   RNLBSERVERID    ded6890'
```

**Pull request merged**